client/state/selectors/get-active-discount.js

```diff
@@ -29,13 +29,22 @@ export const isDiscountActive = ( discount, state ) => {
 		return false;
 	}
 
-	if ( isPlainObject( discount.targetPlan ) ) {
+	if ( _.isPlainObject( discount.targetPlan ) ) {
 		const selectedSiteId = getSelectedSiteId( state );
 		const selectedSitePlanSlug = getSitePlanSlug( state, selectedSiteId );
 
 		return planMatches( selectedSitePlanSlug, discount.targetPlan );
 	}
 
+	if ( _.isArray( discount.targetPlans ) ) {
+		const selectedSiteId = getSelectedSiteId( state );
+		const selectedSitePlanSlug = getSitePlanSlug( state, selectedSiteId );
+
+		return _.some(
```

Comment:
Just curious: is there a reason to use both `some` and `map`? Isn't `some` sufficient to do both?